src/delta.rs

```diff
@@ -317,7 +325,7 @@ fn handle_generic_file_meta_header_line(
     raw_line: &str,
     config: &Config,
 ) -> std::io::Result<()> {
-    if config.file_style.is_omitted {
+    if config.file_style.is_omitted && !config.color_only {
         return Ok(());
     }
```

Comment:
This requires anyway, otherwise we need to refactor `should_handle`.
We can discuss and make another pr if we need.

Reply:
OK. Feel free to add explanatory comments to the code while it is fresh in your brain. If we have some test coverage of this then we'll feel even better about refactoring :)

Reply:
Oh yes! I'm gonna add.

Reply:
@dandavison 
Add comment here def9554